onnxruntime/core/optimizer/gemm_sum_fusion.cc

```diff
@@ -0,0 +1,161 @@
+  // Get currently set attributes of Gemm. Beta will become 1.0.
+  bool transA = static_cast<bool>(gemm_node.GetAttributes().at("transA").i());
+  bool transB = static_cast<bool>(gemm_node.GetAttributes().at("transB").i());
+  float alpha = gemm_node.GetAttributes().at("alpha").f();
+  float beta = 1.0f;
+
+  Node& sum_node = *graph.GetNode(gemm_node.OutputEdgesBegin()->GetNode().Index());
+
+  // Our first two new gemm's input defs are exactly the same as the old gemm's
+  // output defs.
+  std::vector<NodeArg*> new_gemm_input_defs = gemm_node.MutableInputDefs();
+
+  // The other new gemm's input def is the other sum input def.
+  // Also (maybe) get the sum node's other input edge, unless it's a graph input.
+  if (sum_node.InputDefs()[0]->Name() == gemm_node.OutputDefs()[0]->Name()) {
+    new_gemm_input_defs.push_back(sum_node.MutableInputDefs()[1]);
+  } else {
+    new_gemm_input_defs.push_back(sum_node.MutableInputDefs()[0]);
+  }
+  ORT_ENFORCE(new_gemm_input_defs.size() == 3);
+
+  std::vector<NodeArg*> new_gemm_output_defs = sum_node.MutableOutputDefs();
+  ORT_ENFORCE(new_gemm_output_defs.size() == 1);
+
+  Node& new_gemm_node = graph.AddNode(graph.GenerateNodeName(gemm_node.Name() + "_sum_transformed"),
+                                      gemm_node.OpType(),
+                                      "Fused Gemm with Sum",
+                                      new_gemm_input_defs,
+                                      new_gemm_output_defs,
+                                      {},
+                                      gemm_node.Domain());
+  new_gemm_node.AddAttribute("transA", static_cast<int64_t>(transA));
+  new_gemm_node.AddAttribute("transB", static_cast<int64_t>(transB));
+  new_gemm_node.AddAttribute("alpha", static_cast<float>(alpha));
+  new_gemm_node.AddAttribute("beta", static_cast<float>(beta));
+
+  // Move both input edges from original gemm to new gemm.
+  for (auto input_edge : graph_utils::GraphEdge::GetNodeInputEdges(gemm_node)) {
+    ORT_ENFORCE(input_edge.src_arg_index < 2);
+    graph.AddEdge(input_edge.src_node, new_gemm_node.Index(), input_edge.src_arg_index, input_edge.dst_arg_index);
+    graph.RemoveEdge(input_edge.src_node, input_edge.dst_node, input_edge.src_arg_index, input_edge.dst_arg_index);
+  }
+
+  // Move all output edges from sum to new gemm.
+  for (auto output_edge : graph_utils::GraphEdge::GetNodeOutputEdges(sum_node)) {
+    ORT_ENFORCE(output_edge.src_arg_index == 0);
+    graph.AddEdge(new_gemm_node.Index(), output_edge.dst_node, output_edge.src_arg_index, output_edge.dst_arg_index);
+    graph.RemoveEdge(output_edge.src_node, output_edge.dst_node, output_edge.src_arg_index, output_edge.dst_arg_index);
+  }
+
+  // Finally, move the other sum input edge to "C" for the new gemm node.
+  // If The other sum input def is a a graph input, there is no edge to move.
+  bool sum_input_moved = false;
+  for (auto sum_input : graph_utils::GraphEdge::GetNodeInputEdges(sum_node)) {
```

Comment:
```suggestion
  for (auto sum_input_edge : graph_utils::GraphEdge::GetNodeInputEdges(sum_node)) {
```
to make naming consistent with loops above.